modules/setting/setting.go

```diff
@@ -903,9 +903,10 @@ func loadFromConf(allowEmpty bool, extraConfig string) {
 
 	sec = Cfg.Section("security")
 	InstallLock = sec.Key("INSTALL_LOCK").MustBool(false)
-	SecretKey = sec.Key("SECRET_KEY").MustString("!#@FDEWREWR&*(")
 	LogInRememberDays = sec.Key("LOGIN_REMEMBER_DAYS").MustInt(7)
 	CookieUserName = sec.Key("COOKIE_USERNAME").MustString("gitea_awesome")
+	SecretKey = loadSecret(sec, "SECRET_KEY_URI", "SECRET_KEY", generate.NewSecretKey)
```

Comment:
Is it safe to change the default behavior? If some users uses a Gitea upgraded from a old config (without secret setting), or some users just copy the example ini and put an empty `SECRET_KEY=` in config, this change will make them can not read the default secret key `!#@FDEWREWR&*(`, then all data encrypted by the key can not be decrypted (eg: OTP can not be decrypted and users can not login)

Reply:
Hmm, that's actually a really good point; I think that using a default secret key is a really bad idea since it effectively makes the secret key worthless, but we don't want to lock users out without them knowing what's going on.

For this PR, I can change the default generator to just use the default value, but once we implement #16832, we should probably have a migration that moves users off the default key to a properly generated one.